test-data-generator/src/test/java/PlanGeneratorTest.scala

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2019 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import org.scalatest.funsuite.AnyFunSuite
+import za.co.absa.spline.producer.model.v1_2.{DataOperation, ExecutionEvent}
+
+class PlanGeneratorTest extends AnyFunSuite {
```

Comment:
In Spline, we rather prefer using BDD style for writing test. I'm not against `AnyFunSuite` per se, but `AnyFlatSpec` combined with `should.Matchers` offers more natural DSL for writing BDD style tests. Consider using that one. And instead of testing methods (`generateWrite`, `generateRead`) focus on cases (`it should "generate one Write and two Reads when this and that condition is given" in { ... }"`).